ledger/src/blockstore_meta.rs

```diff
@@ -365,6 +412,51 @@ pub struct DuplicateSlotProof {
     pub shred2: shred::Payload,
 }
 
+#[derive(Deserialize, Serialize)]
+pub enum BlockStatus {
+    /// The block is being ingested, of the shreds received there are no conflicts
+    Incomplete,
+
+    /// The block `block_id` has been fully ingested, has consistent shreds and is ready for replay
+    Complete { block_id: Hash },
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct BlockVersions {
+    pub turbine_version: BlockStatus,
+    pub repaired_versions: [(Hash, BlockStatus); 3],
+}
+
+/// Which column an associated block currently resides
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum BlockLocation {
+    Turbine,
+    Repair { block_id: Hash },
```

Comment:
nit: For clarity for column "location", should we rename to `Primary { block_id: Hash }` and `Duplicate { block_id: Hash }`

Reply:
I think, as it is currently, it is clearer what it means, `Turbine` is just the block we receive via block dissemination (that's also why we don't know the `block_id` yet), `Primary` implies that it is the correct block for that slot as decided by consensus, as soon as we know that we would no longer have to store the `Duplicate`s